### ISSUE TITLE
EPERM: operation not permitted, lstat 

### ISSUE THREAD

**Issue by hansfilipelo** [tobyndax/cparts#1]
[Enter steps to reproduce below:]
1. On Ctrl+Alt+O i get the following

Uncaught Error: EPERM: operation not permitted, lstat 'E:\Documents\My Files'

The folder I've opened is E:\Documents and the folder where my source files are is actually in "E:\Documents\src".

**Atom Version**: 1.0.0
**System**: Unknown Windows Version
**Thrown From**: [cparts](https://github.com/tobyndax/cparts) package, v0.1.4
### Stack Trace

Uncaught Error: EPERM: operation not permitted, lstat 'Y:\Documents\My SAS Files'

```
At ATOM_SHELL_ASAR.js:50

Error: EPERM: operation not permitted, lstat 'Y:\Documents\My SAS Files'
    at Error (native)
    at fs.lstatSync (fs.js:836:18)
    at Object.fs.lstatSync (ATOM_SHELL_ASAR.js:170:16)
    at traverse (C:\Users\hael\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\fs-plus\lib\fs-plus.js:370:22)
    at Object.fsPlus.traverseTreeSync (C:\Users\hael\AppData\Local\atom\app-1.0.0\resources\app.asar\node_modules\fs-plus\lib\fs-plus.js:386:14)
    at Object.module.exports.Cparts.changedFile (C:\Users\hael\.atom\packages\cparts\lib\cparts.coffee:199:14)
    at C:\Users\hael\.atom\packages\cparts\lib\cparts.coffee:68:24
    at Pane.module.exports.Pane.observeActiveItem (C:\Users\hael\AppData\Local\atom\app-1.0.0\resources\app.asar\src\pane.js:182:7)
    at Object.module.exports.Cparts.activatePane (C:\Users\hael\.atom\packages\cparts\lib\cparts.coffee:66:39)
    at Object.module.exports.Cparts.toggle (C:\Users\hael\.atom\packages\cparts\lib\cparts.coffee:113:14)
```
### Commands

```
  3x -0:29.6.0 cparts:toggle (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {}
}
```
### Installed Packages

``` coffee
# User
cparts, v0.1.4

# Dev
No dev packages
```

**Comment by tobyndax** [comment]
Reproducable on Windows 8.
Using C:\My Documents; with source and header folders in it. 
Giving EPERM Error, although not on lstat, but on readdir instead.

Cause: fs-plus's traverseTreeSync failing on certain types of folders and files, possibly due to permissions on the folders themselves.

Possbile fix: switch to the asynchronous version in fs-plus.
This is implemented in v0.1.5

Since the reproduced issue doesn't give the exact same error, I'll leave this issue open until hansfilipelo has a chance to test this on his system.

**Comment by hansfilipelo** [comment]
Can't test this right now since I'm no longer on Windows. Will test when I get the time to get around to it.
